design/authenticators/authn_jwt/token_schema.md

```diff
@@ -0,0 +1,180 @@
+[//]: # "2. Design documents should not be updated after implementation"
+[//]: # "3. Design decisions should be made before writing this document, and as such this document should not include options / choices"
+
+
+## Table of Contents
+[//]: # "You can use this tool to generate a TOC - https://ecotrust-canada.github.io/markdown-toc/"
+
+- [Table of Contents](#table-of-contents)
+- [Glossary](#glossary)
+- [Useful Links](#useful-links)
+- [Requirments](#requirments)
+- [Solution](#solution)
+  * [User Interface](#user-interface)
+    + [Authenticator Policy](#authenticator-policy)
+    + [Variable Values Example](#variable-values-example)
+- [Design](#design)
+  * [Class Diagrams](#class-diagrams)
+  * [Flow Explanation](#flow-explanation)
+- [Backwards Compatibility](#backwards-compatibility)
+- [Affected Components](#affected-components)
+- [Delivery Plan](#delivery-plan)
+- [Delivery Plan](#delivery-plan-1)
+- [Open Questions](#open-questions)
+- [Definition of Done](#definition-of-done)
+- [Solution Review](#solution-review)
+
+<small><i><a href='http://ecotrust-canada.github.io/markdown-toc/'>Table of contents generated with markdown-toc</a></i></small>
+
+## Glossary
+
+[//]: # "Describe terms that will be used throughout the design"
+[//]: # "You can use this tool to generate a table - https://www.tablesgenerator.com/markdown_tables#"
+
+| **Term**         | **Description**                                              |
+| ---------------- | ------------------------------------------------------------ |
+| JWT Claims       | claims are pieces of information asserted about a subject    |
+| Host Annotation  | Defining of check that we should do on authentication request for host. In JWT authentication its a claim check. |
+| Mandatory Claims | JWT Claims that host must validate their existance and value |
+| Claims Mapping   | Mapping between claim name and name of host annotation       |
+| Token Schema     | Combination of mandatory claims and claims mapping           |
+
+## Useful Links
+[//]: # "Add links that may be useful for the reader"
+
+| **Name**   | **Link**                                                     |
+| ---------- | ------------------------------------------------------------ |
+| JWT Claims | https://auth0.com/docs/tokens/json-web-tokens/json-web-token-claims |
+
+## Requirments
+[//]: # "Elaborate on the issue you are writing a solution for"
+
+1. Required fields
+   1. The user will have the ability to declare in the authenticator policy which claims must be in host annotations. And therefore also in the JWT token.
+   2. Upon authentication request the authenticator MUST validate those claims - existence and content 
+2. Mapping 
+   1. The user should have the ability to map a given "technical" claim name to a more familiar / business one - i.e, in git-lab - instead of mention in the policy "sub" claim, he will have the ability to mention "job"
```

Comment:
```suggestion
   1. The user should have the ability to map a given "technical" claim name to a more familiar / business one - i.e, in git-lab. Instead of mentioning in the policy "sub" claim, the user will have the ability to mention "job"
```